src/js/components/TableCell/TableCell.js

```diff
@@ -48,13 +48,39 @@ const TableCell = forwardRef(
     const tableContext = useContext(TableContext);
     const cellRef = useForwardedRef(ref);
     const containerRef = useRef();
+    const widthRef = useRef();
 
     useLayoutEffect(() => {
+      let resizeObserver;
+      const element = cellRef.current;
       if (onWidth) {
-        const { width } = cellRef.current.getBoundingClientRect();
-        onWidth(width);
+        if (window?.ResizeObserver) {
```

Comment:
Should we do `typeof window !== 'undefined'` instead of `window?.`
I know we ran into an issue recently with using `window?.` https://github.com/grommet/grommet-icons/pull/349/

Reply:
Great point. Yeah if the variable doesn't exist at all it might be a problem.